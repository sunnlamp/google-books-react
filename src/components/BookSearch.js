import React, { Component } from 'react';
import Form from './Form';
import BookList from './BookList';
import api from '../utils/api';

export default class BookSearch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bookQuery: "",
      books: [],
      errorMessage: "",
      statusMessage: "No books to display, please enter an author or title."
    }
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }


  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({
       [name]: value
    });
  }

  handleFormSubmit = event => {
    event.preventDefault();
    let { bookQuery } = this.state;  
    if (bookQuery !== "") {
      api.getBookData(bookQuery)
      .then(response => response.json())
      .then(data => {
        this.setState({
          books: data.items
        })
      })
      .catch(err => {
        this.setState({
          errorMessage: "The following error has occured: " + err
        });
        console.log(err);
      });
    } else {
      this.setState({
        statusMessage: "Please enter an author or title!"
      });
    }
  }

  render = () => {
    var { books, statusMessage, errorMessage } = this.state;
    return (
      <div className="container">
        {
          errorMessage !== "" ? (
            <h2>{errorMessage}</h2>
          ) : (
            <h2>Google Books API Search</h2>
          )
        }
        <Form 
          bookQuery={this.bookQuery}
          inputChange={this.handleInputChange}
          formSubmit={this.handleFormSubmit}
        />
        {
          books !== null && !books.length ? (
            <h2>{statusMessage}</h2>
          ) : (
            <BookList
              books={books}
            />
          )
        }
      </div>
    )
  }
}
