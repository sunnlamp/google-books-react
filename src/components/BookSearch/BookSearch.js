import React, { Component } from 'react';
import Form from '../Form/Form';
import BookList from '../BookList/BookList';
import api from '../../utils/api';

export default class BookSearch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bookQuery: "",
      books: []
    }
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
    this.handleClearForm = this.handleClearForm.bind(this);
  }


  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleFormSubmit = event => {
    event.preventDefault();
    let { bookQuery } = this.state;
    api.getBookData(bookQuery)
    .then(response => response.json())
    .then(data => {
      this.setState({
        books: data.items
      })
    })
    .then(
      this.handleClearForm()
    )
  }

  handleClearForm = () => {
    this.setState({
      bookQuery: ""
    })
  }

  render = () => {
    var { books } = this.state;
    return (
      <div>
        <Form 
          bookQuery={this.bookQuery}
          inputChange={this.handleInputChange}
          formSubmit={this.handleFormSubmit}
        />
        <div>
          {
            !books.length ? (
              <h1>No books to display, please enter an author or title.</h1>
            ) : (
              <BookList
                books = {books}
              />
          )
          }
        </div>

      </div>
    )
  }
}

// Type in a query and display a list of books matching that query.
// Each item in the list should include the book's author, title, 
// and publishing company, as well as a picture of the book.
// From each list item, you should also be able to navigate to more 
// information about the book, but this information does not necessarily
// need to appear on a page within your application.In other words,
// this could link out to an external site with more information about that particular book.
// items[0].volumeInfo.authors[0] -- authors
// items[0].volumeInfo.title
// items[0].volumeInfo.publisher
// items[0].imageLinks.thumbnail or smallThumbnail
// items[0].canonicalVolumeLink -- link
// items[0].previewLink