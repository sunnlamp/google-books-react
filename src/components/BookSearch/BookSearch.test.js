import React from 'react';
import { mount, shallow } from 'enzyme';
import BookSearch from './BookSearch';

describe('BookSearch', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<BookSearch />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(2);
  });

  it('should display a form with a input', () => {
    expect(wrapper.containsAllMatchingElements([
      <form>
        <input />
      </form>
    ]))
  })
});

describe('mounted BookSearch', () => {
  let wrapper;
  beforeEach(() => wrapper = mount(<BookSearch />));

  // it('calls handleInputChange when a value is entered', () => {
  //   const spy = jest.spyOn(wrapper.instance(), 'handleInputChange');
  //   const input = wrapper.find('input').simulate('click');
  //   input.value = "Harry Potter";
  //   input.simulate('change', input);
  //   expect(spy.called).toEqual(String);
  //   spy.mockRestore();
  // });
});