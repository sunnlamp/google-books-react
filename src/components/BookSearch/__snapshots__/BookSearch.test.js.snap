// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookSearch should render correctly 1`] = `undefined`;
